src/main/daml/Daml/Finance/Instrument/StructuredProduct/AutoCallable/Instrument.daml

```diff
@@ -0,0 +1,155 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.StructuredProduct.AutoCallable.Instrument where
+
+import DA.Date (daysSinceEpochToDate)
+import DA.Set (singleton)
+import Daml.Finance.Data.Reference.HolidayCalendar (getHolidayCalendars, rollSchedule)
+import Daml.Finance.Data.Time.DateClock (dateToDateClockTime)
+import Daml.Finance.Instrument.StructuredProduct.Util (createAutoCallableClaims)
+import Daml.Finance.Interface.Claims.Claim qualified as Claim (GetClaims(..), I, View(..))
+import Daml.Finance.Interface.Claims.Dynamic.Instrument qualified as DynamicInstrument (CreateNewVersion(..), I, View(..))
+import Daml.Finance.Interface.Claims.Types (EventData)
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (I, View(..), createReference, disclosureUpdateReference, instrumentKey)
+import Daml.Finance.Interface.Instrument.StructuredProduct.AutoCallable.Instrument qualified as AutoCallable (I, View(..))
+import Daml.Finance.Interface.Instrument.StructuredProduct.AutoCallable.Types (AutoCallable(..))
+import Daml.Finance.Interface.Types.Common.Types (HoldingStandard(..), Id(..), InstrumentKey(..), PartiesMap)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.Schedule (PeriodicSchedule(..))
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, View(..), flattenObservers)
+import Daml.Finance.Util.Date.Calendar (merge)
+import Daml.Finance.Util.Disclosure (addObserversImpl, removeObserversImpl, setObserversImpl)
+
+-- | Type synonym for `Instrument`.
+type T = Instrument
+
+-- | This template models an AutoCallable instrument that pays a conditional coupon.
+template Instrument
+  with
+    depository : Party
+      -- ^ The depository of the instrument.
+    issuer : Party
+      -- ^ The issuer of the instrument.
+    id : Id
+      -- ^ An identifier of the instrument.
+    version : Text
+      -- ^ The instrument's version.
+    holdingStandard : HoldingStandard
+      -- ^ The holding standard for holdings referencing this instrument.
+    description : Text
+      -- ^ A description of the instrument.
+    referenceAssetId : Text
+      -- ^ The reference asset ID. For example, in case of an AAPL underlying this should be a valid
+      --   reference to the AAPL fixings to be used for the payoff calculation.
+    initialFixing : Decimal
```

Comment:
Do we need to specify this in the instrument? This is typically not known prior to the `fixingDate` and a relative strike level would be specified instead (e.g. 100% of initial fixing) for the put option part.

I don't see a put barrier either, so I assume in this case we have a plain put so that you take exposure as soon as price drops below initial fixing?

Reply:
> I don't see a put barrier either, so I assume in this case we have a plain put so that you take exposure as soon as price drops below initial fixing?

There is a `finalBarrier` to determine the final redemption value, so it is a knock-in put.

Reply:
Ah, I had missed that! Worth specifying then that this is an Autocallable Barrier Reverse Convertible where the KI barrier is observed at maturity